src/public_parameters/disk_cache.rs

```diff
@@ -0,0 +1,63 @@
+use std::fs::{create_dir_all, File};
+use std::io::{BufReader, BufWriter};
+use std::marker::PhantomData;
+use std::path::{Path, PathBuf};
+
+use crate::coprocessor::Coprocessor;
+use crate::proof::nova::{CurveCycleEquipped, PublicParams};
+use crate::public_parameters::error::Error;
+
+pub(crate) struct PublicParamDiskCache<F, C>
+where
+    F: CurveCycleEquipped,
+    C: Coprocessor<F> + 'static,
+{
+    dir: PathBuf,
+    _t: PhantomData<(F, C)>,
+}
+
+impl<F: CurveCycleEquipped, C: Coprocessor<F>> PublicParamDiskCache<F, C> {
+    pub(crate) fn new(disk_cache_path: Option<&Path>) -> Result<Self, Error> {
+        let dir = match disk_cache_path {
+            Some(path) => path.to_owned(),
+            None => {
+                cfg_if::cfg_if! {
+                    if #[cfg(not(target_arch = "wasm32"))] {
+                        home::home_dir()
+                            .expect("missing home directory")
+                            .join(".lurk/public_params")
+                    }
+                    else {
+                        PathBuf::from(".lurk/public_params")
+                    }
+                }
+            }
+        };
```

Comment:
Aha, this is why we're downstreaming an `Option`. Let's make the effort to avoid these dangling hardcodes in deep code. Again, the caller of the topmost API (the one who calls the `public_params` function) should know the placement for cached public parameters